store/errors.go

```diff
@@ -183,28 +191,38 @@ func (e SnapActionError) Error() string {
 			header = "cannot install or download:"
 		}
 	}
-	es := []string{header}
 
-	for name, e := range e.Refresh {
-		es = append(es, fmt.Sprintf("snap %q: %v", name, e))
-	}
+	// reverse the maps, as the common case is that they're all the same
```

Comment:
I would expand the comment a little bit, maybe something like: `reverse the "snap->error" map to "error->snap" because the common case is that all snaps fail with the same error (like no refresh available).`. It took me a bit to get the needed context why its reversed.